src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1121,6 +1132,46 @@ MakeJoinOrderNode(TableEntry *tableEntry, JoinRuleType joinRuleType,
 }
 
 
+/*
+ * IsApplicableJoinClause tests if the current joinClause is applicable to the join at
+ * hand.
+ *
+ *   Given a list of left hand tables and a candidate right hand table the join clause is
+ *   valid if atleast 1 column is from the right hand table AND all columns can be found
+ *   in either the list of tables on the left *or* in the right hand table.
```

Comment:
Seems this has some unintended extra indentation?

Reply:
it was actually intended, albeit not used in our code base often it is often used in postgres.

I will remove the extra indent.